cypress/support/components/ScorerModal.js

```diff
@@ -3,76 +3,116 @@
+export function VerifyContainsDisabledAction(expectedResponse) { VerifyActionState('[data-testid="action-scorer-text-response"]', expectedResponse, '[data-testid="action-scorer-button-no-click"]', true) }
 
-export function ClickEndSessionAction(expectedData) {
-  cy.Get('[data-testid="action-scorer-session-response"]')
-    .ExactMatch('EndSession')
-    .siblings('[data-testid="action-scorer-session-response-user"]')
-    .ExactMatch(expectedData)
-    .parents('div.ms-DetailsRow-fields')
-    .find('[data-testid="action-scorer-button-clickable"]')
-    .Click()
-}
+export function VerifyContainsEnabledEndSessionAction(expectedData) { VerifyActionState('[data-testid="action-scorer-session-response-user"]', expectedData, '[data-testid="action-scorer-button-clickable"]', false) }
+export function VerifyContainsDisabledEndSessionAction(expectedData) { VerifyActionState('[data-testid="action-scorer-session-response-user"]', expectedData, '[data-testid="action-scorer-button-no-click"]', true) }
+export function VerifyContainsSelectedEndSessionAction(expectedData) { VerifyActionState('[data-testid="action-scorer-session-response-user"]', expectedData, '[data-testid="action-scorer-button-selected"]', false) }
 
-export function VerifyContainsEnabledAction(expectedResponse) {
-  cy.Get('[data-testid="action-scorer-text-response"]').contains(expectedResponse)
-    .parents('div.ms-DetailsRow-fields').find('[data-testid="action-scorer-button-clickable"]')
-    .should('be.enabled')
-}
 
-export function VerifyContainsDisabledAction(expectedResponse) {
-  cy.Get('[data-testid="action-scorer-text-response"]').contains(expectedResponse)
-    .parents('div.ms-DetailsRow-fields').find('[data-testid="action-scorer-button-no-click"]')
-    .should('be.disabled')
+// To VALIDATE All of the Data in the Score Actions Grid use this class.
+// 1) Manually verify that the data in each Score Actions grid for each chat turn that the test suite will verify.
+// 2) Generate and persist the data by running the test with the following environment variable:
+//      set CYPRESS_GENERATE_SCORE_ACTIONS_DATA=true
+//    Doing this will generate a file in the "cypress/fixtures/scoreActions" folder that will be used in #3
+// 3) Clear out that environment variable and run the tests as usual. They will grab the generated test data
+//    and use it during validations that use the VerifyScoreActionsList() function.
+//
+// See existing test cases that already use this class as an example to follow.
+
+export class GeneratedData {
+  constructor(dataFileName) {
+    this.dataFileName = dataFileName
+    this.generateScoreActionsData = Cypress.env('GENERATE_SCORE_ACTIONS_DATA')
+
+    if (this.generateScoreActionsData) {
+      this.data = []
+    } else {
+      context('Load Generated Verification Data', () => {
```

Comment:
Would just import this file as any other javascript file.

`import myTestData from './myTestData'`

Then use it as normal. Having all the `cy.readFile` and `cy.fixture` just adds complexity.

Reply:
While I did not realize that would work, and am not sure it really would work, my implementation is different for a reason.

The tests that use these can be run in two ways;
1. Generate mode which generates these data files.
2. Test mode which is the main mode used to test the UI and it consumes these files.

In my opinion, trying to do it the way you suggest would be more complex.